src/backend/libinput.rs

```diff
@@ -0,0 +1,329 @@
+                       handler.on_input_config_changed(&mut self.devices);
+                    }
+                },
+                Event::Touch(touch_event) => {
+                    use ::input::event::touch::*;
+                    if let Some(ref mut handler) = self.handler {
+                        let device_seat = touch_event.device().seat();
+                        trace!(self.logger, "Calling on_touch with {:?}", touch_event);
+                        handler.on_touch(&self.seats.get(&device_seat).expect("Recieved key event of non existing Seat").seat,
+                                         touch_event.time(), touch_event.into())
+                    }
+                },
+                Event::Keyboard(keyboard_event) => {
+                    use ::input::event::keyboard::*;
+                    match keyboard_event {
+                        KeyboardEvent::Key(key_event) => {
+                            if let Some(ref mut handler) = self.handler {
+                                trace!(self.logger, "Calling on_keyboard_key with {:?}", key_event);
+                                let device_seat = key_event.device().seat();
+                                handler.on_keyboard_key(&self.seats.get(&device_seat).expect("Recieved key event of non existing Seat").seat,
+                                                        key_event.time(), key_event.key(), key_event.key_state().into(), key_event.seat_key_count());
+                            }
+                        }
+                    }
+                },
+                Event::Pointer(pointer_event) => {
+                    use ::input::event::pointer::*;
+                    match pointer_event {
+                        PointerEvent::Motion(motion_event) => {
+                            let device_seat = motion_event.device().seat();
+                            let desc = self.seats.get_mut(&device_seat).expect("Recieved pointer event of non existing Seat");
+                            desc.pointer.0 += motion_event.dx() as u32;
+                            desc.pointer.1 += motion_event.dy() as u32;
+                            if let Some(ref mut handler) = self.handler {
+                                trace!(self.logger, "Calling on_pointer_move with {:?}", desc.pointer);
+                                handler.on_pointer_move(&desc.seat, motion_event.time(), desc.pointer);
+                            }
+                        },
+                        PointerEvent::MotionAbsolute(motion_event) => {
+                            let device_seat = motion_event.device().seat();
+                            let desc = self.seats.get_mut(&device_seat).expect("Recieved pointer event of non existing Seat");
+                            desc.pointer = (
+                                motion_event.absolute_x_transformed(
+                                /*FIXME: global.get_focused_output_for_seat(&desc.seat).width() or something like that*/ 1280) as u32,
```

Comment:
I think the easiest and least intrusive way to handle this would be a `set_output_metadata()` method on the `LibinputInputBackend`, that the compositor should call whenever the cursor changes screen, maybe?

Do we know in which cases this kind of event (`MotionAbsolute`) is generated?